build broken with clang 5.0 and Apple's clang 9.0.0

subj

```
Undefined symbols for architecture x86_64:
  "dealii::GeometryInfo<0>::ucd_to_deal", referenced from:
      std::__1::basic_istream<char, std::__1::char_traits<char> >& dealii::DataOutBase::operator>><0, 1>(std::__1::basic_istream<char, std::__1::char_traits<char> >&, dealii::DataOutBase::Patch<0, 1>&) in data_out_base.cc.o
      std::__1::basic_ostream<char, std::__1::char_traits<char> >& dealii::DataOutBase::operator<<<0, 1>(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, dealii::DataOutBase::Patch<0, 1> const&) in data_out_base.cc.o
      std::__1::basic_istream<char, std::__1::char_traits<char> >& dealii::DataOutBase::operator>><0, 2>(std::__1::basic_istream<char, std::__1::char_traits<char> >&, dealii::DataOutBase::Patch<0, 2>&) in data_out_base.cc.o
      std::__1::basic_ostream<char, std::__1::char_traits<char> >& dealii::DataOutBase::operator<<<0, 2>(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, dealii::DataOutBase::Patch<0, 2> const&) in data_out_base.cc.o
      std::__1::basic_istream<char, std::__1::char_traits<char> >& dealii::DataOutBase::operator>><0, 3>(std::__1::basic_istream<char, std::__1::char_traits<char> >&, dealii::DataOutBase::Patch<0, 3>&) in data_out_base.cc.o
      std::__1::basic_ostream<char, std::__1::char_traits<char> >& dealii::DataOutBase::operator<<<0, 3>(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, dealii::DataOutBase::Patch<0, 3> const&) in data_out_base.cc.o
ld: symbol(s) not found for architecture x86_64
```

most likely broken by ~~https://github.com/dealii/dealii/pull/5748~~ https://github.com/dealii/dealii/pull/5745/

@masterleinad any ideas?

@davydden I do not have macOS but compiling on my machine with clang 5.0 results in link failures:

```
../lib/libdeal_II.g.so.9.0.0-pre: error: undefined reference to 'dealii::GeometryInfo<0>::dx_to_deal'
../lib/libdeal_II.g.so.9.0.0-pre: error: undefined reference to 'dealii::GeometryInfo<0>::ucd_to_deal'
clang-5.0: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [examples/CMakeFiles/step-32.debug.dir/build.make:233: bin/step-32.debug] Error 1
```

I can confirm on my mac:

~~~
Undefined symbols for architecture x86_64:
  "dealii::GeometryInfo<0>::ucd_to_deal", referenced from:
      std::__1::basic_istream<char, std::__1::char_traits<char> >& dealii::DataOutBase::operator>><0, 1>(std::__1::basic_istream<char, std::__1::char_traits<char> >&, dealii::DataOutBase::Patch<0, 1>&) in data_out_base.cc.o
      std::__1::basic_ostream<char, std::__1::char_traits<char> >& dealii::DataOutBase::operator<<<0, 1>(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, dealii::DataOutBase::Patch<0, 1> const&) in data_out_base.cc.o
      std::__1::basic_istream<char, std::__1::char_traits<char> >& dealii::DataOutBase::operator>><0, 2>(std::__1::basic_istream<char, std::__1::char_traits<char> >&, dealii::DataOutBase::Patch<0, 2>&) in data_out_base.cc.o
      std::__1::basic_ostream<char, std::__1::char_traits<char> >& dealii::DataOutBase::operator<<<0, 2>(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, dealii::DataOutBase::Patch<0, 2> const&) in data_out_base.cc.o
      std::__1::basic_istream<char, std::__1::char_traits<char> >& dealii::DataOutBase::operator>><0, 3>(std::__1::basic_istream<char, std::__1::char_traits<char> >&, dealii::DataOutBase::Patch<0, 3>&) in data_out_base.cc.o
      std::__1::basic_ostream<char, std::__1::char_traits<char> >& dealii::DataOutBase::operator<<<0, 3>(std::__1::basic_ostream<char, std::__1::char_traits<char> >&, dealii::DataOutBase::Patch<0, 3> const&) in data_out_base.cc.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
~~~

--- a/include/deal.II/base/geometry_info.h
+++ b/include/deal.II/base/geometry_info.h
@@ -1121,7 +1121,7 @@ struct GeometryInfo<0>
    * with the UCD numbering, this field can also be used like a
    * old_to_lexicographic mapping.
    */
-  static constexpr std::array<unsigned int, vertices_per_cell> ucd_to_deal {{0}};
+  static const std::array<unsigned int, vertices_per_cell> ucd_to_deal;
 
   /**
    * Rearrange vertices for OpenDX output.  For a cell being written in OpenDX
@@ -1136,7 +1136,7 @@ struct GeometryInfo<0>
    *   out << cell->vertex(dx_to_deal[i]);
    * @endcode
    */
-  static constexpr std::array<unsigned int, vertices_per_cell> dx_to_deal {{0}};
+  static const std::array<unsigned int, vertices_per_cell> dx_to_deal;
 };
 
 

--- a/source/base/geometry_info.cc
+++ b/source/base/geometry_info.cc
@@ -54,12 +54,11 @@ template <int dim>
 constexpr std::array<unsigned int, GeometryInfo<dim>::vertices_per_cell>
 GeometryInfo<dim>::ucd_to_deal;
 
-constexpr std::array<unsigned int, GeometryInfo<0>::vertices_per_cell>
-GeometryInfo<0>::ucd_to_deal;
+const std::array<unsigned int, GeometryInfo<0>::vertices_per_cell>
+GeometryInfo<0>::ucd_to_deal = {{0}};
 
-constexpr std::array<unsigned int, GeometryInfo<0>::vertices_per_cell>
-GeometryInfo<0>::dx_to_deal;
-
+const std::array<unsigned int, GeometryInfo<0>::vertices_per_cell>
+GeometryInfo<0>::dx_to_deal = {{0}};
 
 template struct GeometryInfo<1>;
 template struct GeometryInfo<2>;

